packages/typescript-estree/src/ts-estree/ts-estree.ts

```diff
@@ -521,7 +521,7 @@ interface ClassDeclarationBase extends BaseNode {
 
 /** this should not be directly used - instead use ClassPropertyComputedNameBase or ClassPropertyNonComputedNameBase */
 interface ClassPropertyBase extends BaseNode {
-  key: PropertyNameComputed | PropertyNameComputed;
```

Comment:
this is invalid, it should be `PropertyNameComputed | PropertyNameNonComputed` = PropertyName